tsl/test/isolation/specs/cagg_concurrent_invalidation.spec

```diff
@@ -90,4 +112,17 @@ step "s3_check_watermarks" {
       ORDER BY mat_hypertable_id;
 }
 
+# Check that all watermarks in all sessions are up to date after running an update 
 permutation "s3_lock_invalidation" "s1_run_update" "s2_run_update" "s3_release_invalidation" "s3_check_watermarks"("s1_run_update", "s2_run_update")
+
+###
+# Check that we don't leak the old AND the new value of the invalidation watermark during an update into another session
+###
+
+# (1) Session 2 - Inserts new data of the year 2022 to give the refresh job some work to do
+# (2) Session 3 - Blocks the invalidation watermark scanner after reading the first watermark for the hypertable
+# (3) Session 2 - Inserts new data of the year 2023, during the commits it reads the current invalidation watermark and blocks due to (2)
+# (4) Session 1 - Refreshes a CAgg on this hypertable. During the update the invalidation watermark is deleted and a new one is written
+# (5) Session 3 - Unblocks the scanner of session 2
+# (6) Session 2 - Continues the scan. The new inserted invalidation watermark should not be seen since the transaction was started after s2 has started the scan in (3)
```

Comment:
I'm a little confused about this part -- if session 2 does a backfill into the range that was refreshed by session 1, what will be the final watermark value?